crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -548,17 +548,33 @@ async fn get_quote(
         // verified quote here on purpose.
         verification: Default::default(),
     };
-    let fee_amount = match order_data.fee_amount.is_zero() {
-        // If an ETHFlow order was created with 0 fee it means it has the semantics
-        // of a limit order where solvers are responsible of baking in an appropriate execution fee
-        // into their limit prices. In that case we don't require that the fee amount matches one of
-        // our stored quotes.
-        true => None,
-        false => Some(order_data.fee_amount),
-    };
-    get_quote_and_check_fee(quoter, &parameters.clone(), Some(*quote_id), fee_amount)
-        .await
-        .map_err(onchain_order_placement_error_from)
+    let mut result = get_quote_and_check_fee(
+        quoter,
+        &parameters.clone(),
+        Some(*quote_id),
+        Some(order_data.fee_amount),
+    )
+    .await;
+
+    // If we didn't find the quote, recompute a fresh one
```

Comment:
but didn't previous call of `get_quote_and_check_fee` already tried to recompute a fresh one if it didn't find the specified one?

Reply:
Only if no quote id was specified (which for ETH flow orders is not possible): https://github.com/cowprotocol/services/blob/b8161a5ac8530f7b35861482208bcea40f6bdb52/crates/shared/src/order_validation.rs#L858

If a quote id was specified it fails with "Not Found"